src/kernel/include/isb.h

```diff
@@ -0,0 +1,46 @@
+#ifndef ISB_H
+#define ISB_H 
+
+#include "component.h"
+#include "cap_ops.h"
+#include "pgtbl.h"
+
+struct cap_isb {
+    struct cap_header     h;
+    struct liveness_data  liveness;
+    vaddr_t               kern_addr;
+} __attribute__((packed));
+
+static int
+isb_activate(struct captbl *t, capid_t ctcap, capid_t isbcapid, livenessid_t lid, vaddr_t kaddr, capid_t ptcid, vaddr_t uaddr)
+{
+    struct cap_isb   *isbcap;
+    struct cap_pgtbl *ptcap;
+    paddr_t           pframe = chal_va2pa((void *)kaddr);
+    int               ret;
+
+    ptcap = (struct cap_pgtbl *)captbl_lkup(t, ptcid);
+    if (!ptcap || ptcap->h.type != CAP_PGTBL) return -EINVAL;
+    if (pgtbl_mapping_add(ptcap->pgtbl, uaddr, pframe, PGTBL_USER_DEF, 12)) return -EINVAL;
+
```

Comment:
This might be a problem with DCB/SCBs as well, but there is no retyping happening here. So that means that the kernel will believe the memory for this is virtual memory, not kernel memory. I'm not sure what problems could come of this, but I'd expect there would be some.

Reply:
I believe that `cap_kmem_activate` (in capinv.c) performs the retyping when the memory is activated.